docs/reference/data-streams/tsds.asciidoc

```diff
@@ -0,0 +1,298 @@
+
+[discrete]
+[[time-bound-indices]]
+==== Time-bound indices
+
+In a TSDS, each backing index, including the write index, has a range of
+accepted `@timestamp` values. This range is defined by the
+<<index-time-series-start-time,`index.time_series.start_time`>> and
+<<index-time-series-end-time,`index.time_series.end_time`>> index settings.
+
+When you add a document to a TSDS, {es} adds the document to the appropriate
+backing index based on its `@timestamp` value. As a result, a TSDS can add
+documents to any TSDS backing index that can receive writes. This applies even
+if the index isn't the current write index.
+
+image::images/data-streams/time-bound-indices.svg[align="center"]
+
+TIP: Some {ilm-init} actions, such as <<ilm-forcemerge,`forcemerge`>>,
+<<ilm-shrink,`shrink`>>, and <<ilm-searchable-snapshot,`searchable_snapshot`>>,
+make a backing index read-only. You cannot add documents to read-only indices.
+Keep this in mind when defining the index lifecycle policy for your TSDS.
+
+If no backing index can accept a document's `@timestamp` value, {es} rejects the
+document.
+
+
+{es} automatically configures `index.time_series.start_time` and
+`index.time_series.end_time` settings as part of the index creation and rollover
+process.
+
+[discrete]
+[[tsds-look-ahead-time]]
+==== Look-ahead time
+
+{es} uses the <<index-look-ahead-time,`index.look_ahead_time`>> index setting to
+configure the `@timestamp` ranges for a TSDS's write index. When you create a
+new write index for a TSDS, {es} calculates the index's
+`index.time_series.end_time` value as:
+
+`now + index.look_ahead_time`
+
+
+At the <<indices-lifecycle-poll-interval,`indices.lifecycle.poll_interval`>>,
+{es} checks if the write index has met the rollover criteria in its index
+lifecycle policy. If not, {es} refreshes the `now` value and updates the write
+index's `index.time_series.end_time` to:
+
+`now + index.look_ahead_time + indices.lifecycle.poll_interval`
+
+This process continues until the write index rolls over. When the index rolls
+over, {es} sets a final `index.time_series.end_time` value for the index. This
+value borders the `index.time_series.start_time` for the new write index. This
+ensures the `@timestamp` ranges for neighboring backing indices always border
+but never overlap.
+
+[discrete]
+[[dimension-based-routing]]
+==== Dimension-based routing
+
+Within each TSDS backing index, {es} uses the
+<<index-routing-path,`index.routing_path`>> index setting to route documents
+with the same dimensions to the same shards.
+
+When you create the matching index template for a TSDS, you must specify one or
+more dimensions in the `index.routing_path` setting. Each document in a TSDS
+must contain one or more dimensions that match the `index.routing_path` setting.
+
+Dimensions in the `index.routing_path` setting must be plain `keyword` fields.
+The `index.routing_path` setting accepts wildcard (`*`) patterns and can
```

Comment:
Maybe use `dim.*` as an example instead of `*` - `*` would match all fields in a probably janky way.

Reply:
Changed to:
```
Dimensions in the `index.routing_path` setting must be plain `keyword` fields.
The `index.routing_path` setting accepts wildcard patterns (for example `dim.*`)
and can dynamically match new fields.
```